### PR TITLE
lib/cmwx1: Add CMWX1ZZABZ-093 driver.

### PR DESCRIPTION
**Pull request by iabdalkader**

### CODE REVIEW

File: lib/cmwx1/cmwx1.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+        sleep_ms(200)
+        self.rst_pin.value(0)
+        sleep_ms(200)
+        self.rst_pin.value(1)
+
+        # Restart module
+        self.restart()
+
+    def init_modem(self):
+        # Portenta vision shield configuration
```
Comment (sebromero) [inline lib/cmwx1/cmwx1.py:109]:
```suggestion
        # Arduino Portenta H7 Pin Configuration
```

File: lib/cmwx1/cmwx1.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+BAND_AU915 = 1
+BAND_EU868 = 5
+BAND_KR920 = 6
+BAND_IN865 = 7
+BAND_US915 = 8
+BAND_US915_HYBRID = 9
+
+CLASS_A = "A"
+CLASS_B = "B"
+CLASS_C = "C"
```
Comment (sebromero) [inline lib/cmwx1/cmwx1.py:31]: Let's scope them under the Lora class as you suggested to avoid an import like `from cmwx1 import *`

**Review by sebromero — CHANGES_REQUESTED** [review]

**Comment by iabdalkader** [comment]
> Shall we also get rid of the `pyb` and `utime` imports and use the new ones from machine and time?

Done, all other comments fixed too. Note we can't use `__init__.py` here due to how this will be frozen, but it doesn't change how it's used, i.e., you can `from cmwx1 import *` or import individual names, and constants are under `Lora.*`

File: lib/cmwx1/cmwx1.py [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+        "+ERR_RX": LoraErrorRX,
+        "+ERR_UNKNOWN": LoraErrorUnknown,
+    }
+
+    def __init__(
+        self,
+        uart=None,
+        rst_pin=None,
+        boot_pin=None,
+        band=5,
```
Comment (sebromero) [inline lib/cmwx1/cmwx1.py:86]:
```suggestion
        band=None,
```

File: lib/cmwx1/cmwx1.py [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+        poll_ms=300000,
+        debug=False,
+    ):
+        self.debug = debug
+        self.uart = uart
+        self.rst_pin = rst_pin
+        self.boot_pin = boot_pin
+        self.band = band
```
Comment (sebromero) [inline lib/cmwx1/cmwx1.py:94]:
```suggestion
        if band is None:
            band = self.BAND_EU868 
        else:
            self.band = band
```

**Review by sebromero — CHANGES_REQUESTED** [review]
I suggest we use the None pattern instead of hardcoding the band constant in the init function.

File: lib/cmwx1/cmwx1.py (hunk 3 above)
Comment (iabdalkader) [reply lib/cmwx1/cmwx1.py]: I reverted the constant, was removed to fix the linter.

**Pull request merged**